internal/service/advancedclustertpf/resource.go

```diff
@@ -129,20 +129,26 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
 	if diags.HasError() {
 		return
 	}
-	var stateAdvConfig *types.Object
-	if advConfigChanged || cluster == nil {
-		// Cluster changes to UPDATING state after updating advanced configuration
-		cluster = AwaitChanges(ctx, r.Client.AtlasV2.ClustersApi, &plan.Timeouts, diags, plan.ProjectID.ValueString(), plan.Name.ValueString(), changeReasonUpdate)
+	modelOut := &state
+	if clusterResp != nil {
+		modelOut, _ = getBasicClusterModel(ctx, diags, r.Client, clusterResp, &plan)
 		if diags.HasError() {
 			return
 		}
-	} else {
-		stateAdvConfig = &state.AdvancedConfiguration
 	}
-	model := r.convertClusterAddAdvConfig(ctx, legacyAdvConfig, advConfig, cluster, &plan, stateAdvConfig, diags)
-	if model != nil {
-		diags.Append(resp.State.Set(ctx, model)...)
+	if advConfigChanged {
+		_ = AwaitChanges(ctx, r.Client.AtlasV2.ClustersApi, &plan.Timeouts, diags, plan.ProjectID.ValueString(), plan.Name.ValueString(), changeReasonUpdate)
```

Comment:
Why discard the result? This will be an updated clusterResp

Reply:
@EspenAlbert uhm, what information could come from here that we don't already have?

Reply:
applyAdvancedConfigurationChanges returns advConfig and legacyAdvConfig that are enough for setting adv_config attributes.

for the rest of changes we have a good clusterResp from applyTenantUpgrade or applyClusterChanges, or the state if there was not any change.

in fact I've moved it inside applyAdvancedConfigurationChanges to wait as we do in applyTenantUpgrade and applyClusterChanges, WDYT?  in this way, all apply* funcs wait until the changes are done.

commit: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2865/commits/dbd47c6e44c1f68819024547dee6b8cf8e97d163

Reply:
i've added a wait aftet each advanced config update (latest and legacy) in case there is any concurrency issue when updating clusters: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2865/commits/52db86441e2e88a211b1dd8074d3b6aee81b5bf0

Reply:
> uhm, what information could come from here that we don't already have?
Just, ensuring we use the latest read version, don't see a reason not too. I like the other changes 👍